Remove gevent tests and prepare support for es 2.x

ael-code opened the pull request.

boyska commented:
great, this works as expected.

ael-code committed:
conf: removed `defaults` parameter from load_settings func

the function do not make use of it.

ael-code committed:
conf: do not ignore exception while reading configuration file

substituted both logging.warning with a unified well written exception.
One test was changed.

ael-code committed:
cli: make use of load_cfg wrapper

using the wrapper we have cleaner code and
consitent behaviour among all commands

ael-code committed:
cli: expose utility function die() to all commands

ael-code committed:
web gui: drop usage of js to make list item clickable

on `search.html` and `recents.html` pages we were using javascript
to make item of list of the results clickable.

This was causing some problems:
 - right click on div did not show any link
 - ctrl-click did not open another page on firefox
 - ctrl-clcik on the title was opening two pages :)

Now we've drop js for this job and we simply wrap the item <div> with an
<a> block.

ael-code committed:
travis: remove gevent tests and support for es 2.x

ael-code commented:
How do we patch/bypass the elasticsearch version in setup.py in order to test with ealsticsaerch>=2.0 ?

boyska commented:
I have many reasons to refuse this PR:
1. there are already too many pending pull requests about travis, and this is patching a pull request opened 4 months ago
2. one of those pull requests insomnia-lab/libreant#263 is already about better elasticsearch testing
3. I guess (but might be wrong) that `sudo: true` is moving us away of the container-based travis infrastructure. I think that this might have consequences, but I might have been confused by old documentation

More generally, I don't see the hurry about testing elasticsearch2 when we still need to improve testing about so many other things. If you have reasons to do that so early, please explain.

As for your question, I think that `sed` is the right command :)

ael-code committed:
webant: enable pagination on search page.

ael-code committed:
wrap pagination template inside a jinjia macro

ael-code committed:
Merge pull request #225 from boyska/travis-matrix

Travis matrix

ael-code committed:
grant read capability to volumes to anonymous user by default

ael-code committed:
pagination: added support for rel=(prev|next)

This allows bot to understand which is the following page
Additional details can be found here:
https://support.google.com/webmasters/answer/1663744

ael-code committed:
expose RESULTS_PER_PAGE default configuration

ael-code committed:
changelog: v0.4

boyska committed:
Merge pull request #269 from ael-code/pagination

Pagination

ael-code committed:
bump to v0.4

boyska committed:
Merge pull request #271 from insomnia-lab/release-0.4

Release 0.4

ael-code committed:
conf: log path of conf file with bad json format

ael-code committed:
conf: test file with bad json format

ael-code committed:
fix Fsdb library version to 0.3.3

boyska committed:
Merge pull request #276 from ael-code/pyfsdb_version

force Fsdb library version to 0.3.3

refs #254

ael-code committed:
extend fsdb compatibility till v1.2.1

leophys committed:
Merge pull request #270 from ael-code/anonymous_read_capability

Merged!

boyska committed:
Merge pull request #277 from ael-code/pyfsdb_version

extend fsdb compatibility till v1.2.1

ael-code committed:
Merge branch 'dev' into conf-file-failures

leophys committed:
Merge pull request #260 from ael-code/conf-file-failures

Closed.
